### ISSUE TITLE
Improve commands help messages

### ISSUE THREAD

**Issue by jooola** [hetznercloud/cli#708]
### TL;DR

The following two --help outputs have inconsistent usage description:
```
Usage:
  hcloud server attach-to-network [FLAGS] SERVER

Usage:
  hcloud network add-subnet NETWORK FLAGS
```

Order of flags and arguments should not matter, but for consistency, we should always propose the same format in the help output.

Another details, is that sometimes we use `[FLAGS]` with brackets marking it as optional, and sometimes not. But according to the further output of `hcloud server attach-to-network --help` the flags here are not optional either, it lists `--network` as a required flag.

We might also want to stop yelling at our users, and not use UPPERCASE keywords unless really necessary.

### Expected behavior

We could use the http://docopt.org/ flags and arguments description language to follow a well known convention.

Related to #700
